GNUmakefile

```diff
@@ -3,6 +3,9 @@ GOFMT_FILES?=$$(find . -name '*.go' |grep -v vendor)
 WEBSITE_REPO=github.com/hashicorp/terraform-website
 PKG_NAME=random
 
+.EXPORT_ALL_VARIABLES:
+GOFLAGS=-mod=vendor
```

Comment:
I wasn't sure myself on enforcing the use of the vendor folder for local development. I would say the `vendoring` is a relic of speeding up builds for CI in a pre go cache world. I had been actually enforcing modules on just in Travis. It's something we likely want to harmonize but we can figure that out another day, I'm happy to merge anything that "just works"

Reply:
Yes, I agree vendoring is a bigger discussion. At the moment, most providers are using vendoring, at least on TeamCity where it is used for nightly test and release builds. I think we should try to make the development environment match the test/release environment as much as possible, which means that for the moment while vendoring is still being used in TC, it should be used for local development. Since Travis uses the Makefile, setting `GOFLAGS` here brings both local dev and PR test environments into alignment with TC.

If we remove vendoring in the future, we will have to touch either the Makefile or `.travis.yml`, wherever we have set `GOFLAGS=-mod=vendor` (the build will error if there is no `vendor/` directory, so we can't leave it in harmlessly).

What do you think?

Reply:
In fact, I think I've persuaded myself: whatever we decide to do about vendoring, it's not really related to this PR. Will remove this commit.

Reply:
Agree with how this shook out, but wanted to know that our official internal guidance at the moment is to use vendoring when using Go Modules, and deciding against doing that should probably be an RFC.